src/diagnostics.ts

```diff
@@ -409,26 +416,32 @@ async function queryCrateDiagnostics(prusti: dependencies.PrustiLocation, rootPa
  * @returns An array of diagnostics for the given rust project.
  */
 async function queryProgramDiagnostics(prusti: dependencies.PrustiLocation, programPath: string, serverAddress: string, destructors: Set<util.KillFunction>): Promise<[Diagnostic[], VerificationStatus, util.Duration]> {
-    // For backward compatibility
+    const prustiRustcArgs = [
+        "--crate-type=lib",
+        "--error-format=json",
+        "--edition=2018",
+        programPath
+    ].concat(
+        config.extraPrustiRustcArgs()
```

Comment:
Concatenating these after the `programPath` is a bit strange but seems to work just fine when I test it.
We should keep the two `--crate-type=lib` and `--error-format=json` as hardcoded, but could we not move `--edition=2018` as a default setting of `extraPrustiRustcArgs`?